app/lib/pre_assembly/assembly_directory.rb

```diff
@@ -4,20 +4,22 @@ module PreAssembly
   # Represents the assembly structure on the filesystem,
   # Used by PreAssembly::DigitalObject
   class AssemblyDirectory
-    def self.create(druid_id:)
-      new(druid_id: druid_id).tap(&:create_object_directories)
+    def self.create(druid_id:, common_path: '')
+      new(druid_id: druid_id, common_path: common_path).tap(&:create_object_directories)
     end
 
-    def initialize(druid_id:)
+    def initialize(druid_id:, common_path: '')
       @druid_id = druid_id
+      @common_path = common_path
     end
 
     # @return [String] the appropriate path for the file ('metadata' or 'content' will be last segment)
     def path_for(item_path)
       # these are the names of files that will be staged in the 'metadata' folder instead of the 'content' folder
       metadata_files = ['descMetadata.xml', 'contentMetadata.xml'].map(&:downcase)
 
-      return content_dir unless metadata_files.include?(File.basename(item_path).downcase)
+      relative_path = item_path.gsub(common_path, '')
+      return File.join(content_dir, relative_path) unless metadata_files.include?(File.basename(item_path).downcase)
 
```

Comment:
This is a critical change... previously the path for any staged file was simply the "content" folder within the staged object folder.  This would just flatten any incoming hierarchy.   By passing in the common path to this object of the files being staged, we can preserve the hierarchy when staging.